dumper/lib-ext-hive-metastore/src/main/java/com/google/edwmigration/dumper/ext/hive/metastore/ThriftJsonSerializer.java

```diff
@@ -32,4 +35,19 @@ public ThriftJsonSerializer() throws TTransportException {
   public String serialize(TBase<?, ?> thriftObject) throws TException {
     return underlyingSerializer.toString(thriftObject);
   }
+
+  public void serialize(List<? extends TBase<?, ?>> thriftObjects, Writer writer)
```

Comment:
Lock is not required for `Writer` in general. It's a choice at the call-site, whether to obtain one.

I don't want to return any type because of the suggestion to push the serialization to the last possible moment, so that the serialized collection does not accumulate in memory, but is serialized one by one before being passed to the output (in this case to `Writer`). See this comment in another PR: https://github.com/google/dwh-migration-tools/pull/403#discussion_r1617981895

Reply:
Wouldn't calling this method without a lock cause a possibility for outputs to be malformed when two threads are interleaving their `writer` usages?

Reply:
Yes. Therefore in the call-site, if the `Writer` is used from a single thread only, there is no need to think about mutual exclusion (e.g. using locks). If it is used from multiple threads, then mutual exclusion should be guaranteed before calling this method.

Reply:
I've added a comment that the method is not thread-safe. Thanks for pointing that out.